bin/genome-env-next

```diff
@@ -0,0 +1,432 @@
+#!/bin/bash
+
+# if this was found via PATH as opposed to a path to executable
+THIS="$(basename "$0")"
+if test "$THIS" == "$0"
```

Comment:
The infinite re-exec protection suggested below works to do what I wanted to do here as well, i.e. have an ability to disable the re-exec.